### PR TITLE
Improve sdk

### PR DESCRIPTION
**Pull request by boyeon555**
- Revise the definition of required and optional at part of Transactions on Readme same as JSON-RPC API Readme
- Revise the property name of `signed_transaction_dict` to `signed_transaction_as_dict` on SignedTransaction
- Amend the method name of `to_dict` to 'convert_tx_to_jsonrpc_request' to distinguish with the method of 'to_dict' on the 'Transaction' classes
- Make the call and transaction objects have the method `to_dict` to convert the dict to call or transaction object
  - Also supports for the opposite case
  - Add test cases
- Add raising DataTypeException on the method of `convert_params_value_to_hex_str`
- Remove the check if message and call transaction does not have the value field
  - Message and call transaction can have the value field. On the other side, deploy transaction should not have the value field.
- Support for the adding of tbears commands by not loading tests directory on the method 'gen_deploy_data_content'
  - Add test score of 'sample_token' having the tests directory
  - Add unit tests
- Update requests on setup.cfg

### CODE REVIEW

File: iconsdk/signed_transaction.py [hunk 1]
```diff
@@ -31,31 +31,27 @@ def __init__(self, transaction: Transaction, wallet: Wallet):
...
         """
-        self.__signed_transaction_dict = self.to_dict(transaction, wallet)
+        self.__signed_transaction_dict = self.convert_tx_to_jsonrpc_request(transaction, wallet)
         message_hash = sha3_256(serialize(self.__signed_transaction_dict)).digest()
         signature = wallet.sign(message_hash)
         self.__signed_transaction_dict["signature"] = b64encode(signature).decode()
 
     @property
-    def signed_transaction_dict(self):
+    def signed_transaction_as_dict(self):
```
Comment (eunsoo-icon) [inline iconsdk/signed_transaction.py]: Does it need to change the name of signed_transaction_dict property?
When using signed_transaction_dict in a program, updating the icon-python-sdk causes the program to fail and not work.

Comment (boyeon555) [reply iconsdk/signed_transaction.py]: I missed that point. To make the meaning of the property of `signed_transaction_dict` clarify, I tried to change its name. And I've thought there are few users using it.

Comment (boyeon555) [reply iconsdk/signed_transaction.py]: It reverted to `signed_transaction_dict`.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+language: python
+python:
+- "3.6"
+
+install:
+- git clone "https://oauth2:$Gitlab_token@repo.theloop.co.kr/icon/icon-sdk/icon-sdk-java-test.git"
```
Comment (whylee259) [inline .travis.yml]:
```suggestion
- git clone "https://oauth2:$Gitlab_token@repo.theloop.co.kr/icon/icon-sdk/icon-mock-node.git"
```

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+- cd icon-sdk-java-test
```
Comment (whylee259) [inline .travis.yml]:
```suggestion
- cd icon-mock-node
```

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+- npm install
+- node index.js &
+- sleep 5
+- cd ..
+
+script:
+- python setup.py test
+
+before_deploy:
+- rm -rf icon-sdk-java-test
```
Comment (whylee259) [inline .travis.yml]:
```suggestion
- rm -rf icon-mock-node
```

**Pull request merged**